app/models/ability.rb

```diff
@@ -132,6 +141,10 @@ def initialize(user)
 
   private
 
+  def self.active_pro_subscription?(user)
+    user.pro_account && user.pro_account.active?
```

Comment:
I think we can just look for the `pro` role here, otherwise we'll be hammering Stripe with API calls.

Why do we need to check the `pro_account` status?

* When you become a pro, you get the `pro` role applied.
* When your subscription expires, you get the `pro` role removed (this isn't implemented yet, but in progress in https://github.com/mysociety/alaveteli-professional/issues/371).

Reply:
I based this on the same logic that we're currently using to determine whether you can see the dashboard which is to check `pro_account.active?` (which looks for the `stripe_customer` and tests for active Stripe subscriptions - obviously this was written before we realised we might have a clash with other apps - is this not correct?

As an interesting counter example - my main WDTK user has the pro role but not dashboard access

Reply:
wait, that's not quite right - I have dashboard but get bounced out of `SubscriptionsController#index`

Reply:
I've had a look and can't find any use of `ProAccount#active?` to control access to the dashboard.

We use it in two places, where we're directly making use of Stripe objects:

* [PlansController](https://github.com/mysociety/alaveteli/blob/dacbdd3/app/controllers/alaveteli_pro/plans_controller.rb#L38)
* [SubscriptionsController](https://github.com/mysociety/alaveteli/blob/dacbdd3/app/controllers/alaveteli_pro/subscriptions_controller.rb#L146)

These were added because at the time, some users had the `pro` role, but no `Stripe::Subscription`. This was a slightly different use case to the normal permission access, as in these two situations we were letting someone modify their access (either by signing up for an account, or landing on the page that allowed them to cancel). Most pro features rely on the `pro` role being assigned.

Now that _all_ users with the `pro` role have subscriptions, I don't think we need to check against Stripe in either of these cases.

Reply:
> I have dashboard but get bounced out of SubscriptionsController#index
 
Your user doesn't have a `pro_account`, so you wouldn't have been picked up by the [backloading script](https://github.com/mysociety/alaveteli-professional/issues/382#issuecomment-339703490):

```ruby
User.pro.map { |u| u.name if u.pro_account.nil? }.compact
# => ["Liz Conlan"]
```

You'll need to follow the [stripped down script](https://github.com/mysociety/alaveteli-professional/wiki/Pro-Account-Request-Response-Script) to create yourself a `pro_account` record and a stripe customer.

Reply:
cool, I'll fix my account eventually (my fault tbh), will rewrite (simplify) the checks based on the pro role

Reply:
Feel free to force-push these. I haven't reviewed enough for it to matter, and it will be easier to just assess with the updated logic given it will be a pretty big change.

Reply:
Or maybe just make a new PR – whichever works :)